climpred/classes.py

```diff
@@ -0,0 +1,197 @@
+            dimensions `time` and `initialization` at the minimum.""")
+
+
+def _check_reference_dimensions(init, ref):
+    """Checks that the reference matches all initialized dimensions except
+    for 'time'"""
+    init_dims = list(init.dims)
+    if 'time' in init_dims:
+        init_dims.remove('time')
+    if not (set(ref.dims) == set(init_dims)):
+        raise ValueError("""Reference dimensions must match initialized
+            prediction ensemble dimensions (excluding `time`.)""")
+
+
+# TODO: Don't force same variables for every reference. Allow for some
+# references to only have a subset of the initialized prediction variables.
+# This means skill, etc. will be computed for whatever variables
+# are available.
+def _check_reference_vars_match_initialized(init, ref):
+    """
+    Checks that a new reference (or control) dataset has identical variables
+    to the initialized dataset. This ensures that they can be compared
+    pairwise.
+
+    ref: new addition
+    init: dp.initialized
+    """
+    init_list = [var for var in init.data_vars]
+    ref_list = [var for var in ref.data_vars]
+    if set(init_list) != set(ref_list):
+        raise ValueError("""Please provide a dataset with matching variables
+        (and variable names) to the initialied prediction ensemble.""")
+
+
+def _check_xarray(xobj):
+    if not isinstance(xobj, (xr.Dataset, xr.DataArray)):
+        raise ValueError("""You must input an xarray Dataset or DataArray.""")
+
+
+# ----------
+# Aesthetics
+# ----------
+def _display_metadata(self):
+    header = f'<climpred.{type(self).__name__}>'
+    summary = header + '\nInitialized Ensemble:\n'
+    summary += '    ' + str(self.initialized.data_vars)[18:].strip()
+    if isinstance(self, ReferenceEnsemble):
+        # TODO: convert to decorator
+        if any(self.reference):
+            for key in self.reference:
+                summary += f'\n{key}:'
+                N = len(self.reference[key].data_vars)
+                for i in range(1, N+1):
+                    summary += '\n    ' + \
+                               str(self.reference[key].data_vars) \
+                               .split('\n')[i].strip()
+        else:
+            summary += '\nReferences:\n'
+            summary += '    None'
+    elif isinstance(self, PerfectModelEnsemble):
+        summary += '\nControl:\n'
+        # TODO: convert to decorator
+        if any(self.control):
+            N = len(self.control.data_vars)
+            for i in range(1, N+1):
+                summary += '    ' + \
+                           str(self.control.data_vars) \
+                           .split('\n')[i].strip() + '\n'
+        else:
+            summary += '    None'
+    return summary
+
+
+# -----------------
+# CLASS DEFINITIONS
+# -----------------
+class PredictionEnsemble:
+    def __init__(self, xobj):
+        _check_xarray(xobj)
+        if isinstance(xobj, xr.DataArray):
```

Comment:
like the conversion. default will be working with datasets

Reply:
Yes this just forces us to always expect datasets. So whether there is one variable there or ten it will be the same infrastructure.